File: Phoenix/src/main/java/com/niit/mks/controller/AdminController.java
package com.niit.mks.controller;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.niit.mks.dao.BlogDAO;
import com.niit.mks.dao.FriendDAO;
import com.niit.mks.dao.UserDAO;
import com.niit.mks.model.Blog;
import com.niit.mks.model.BlogListModel;
import com.niit.mks.model.Friend;
import com.niit.mks.model.User;
import com.niit.mks.service.EmailService;

@RestController
public class AdminController {

	@Autowired
	Blog blog;

	@Autowired
	BlogDAO blogDAO;

	@Autowired
	User user;

	@Autowired
	UserDAO userDAO;

	@Autowired
	Friend friend;

	@Autowired
	FriendDAO friendDAO;

	@Autowired
	private EmailService emailService;

	public AdminController() {
		System.out.println("Instantiating AdminController");
	}

	@RequestMapping(value = "/admin/blogs", method = RequestMethod.GET)
	public ResponseEntity<List<BlogListModel>> listAllblogs() {
		List<Blog> blogs = blogDAO.getblogsByStatus("PENDING");
		List<BlogListModel> bloglist = new ArrayList<BlogListModel>();

		BlogListModel blogModel = null;

		for (Blog b : blogs) {
			blogModel = new BlogListModel();
			blogModel.setBlog(b);
			blogModel.setFirstName(userDAO.getById(b.getUserId()).getFirstName());
			blogModel.setLastname(userDAO.getById(b.getUserId()).getLastName());
			bloglist.add(blogModel);

		}

		if (bloglist.isEmpty()) {
			blog = new Blog();
			blog.setErrorCode("404");
			blog.setErrorMessage("No blogs present.");
			blogs.add(blog);
		}
		return new ResponseEntity<List<BlogListModel>>(bloglist, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/approvedBlogs", method = RequestMethod.GET)
	public ResponseEntity<List<BlogListModel>> listApprovedblogs() {
		List<Blog> blogs = blogDAO.getblogsByStatus("APPROVE");
		List<BlogListModel> bloglist = new ArrayList<BlogListModel>();

		BlogListModel blogModel = null;

		for (Blog b : blogs) {
			blogModel = new BlogListModel();
			blogModel.setBlog(b);
			blogModel.setFirstName(userDAO.getById(b.getUserId()).getFirstName());
			blogModel.setLastname(userDAO.getById(b.getUserId()).getLastName());
			bloglist.add(blogModel);

		}

		if (bloglist.isEmpty()) {
			blog = new Blog();
			blog.setErrorCode("404");
			blog.setErrorMessage("No blogs present.");
			blogs.add(blog);
		}
		return new ResponseEntity<List<BlogListModel>>(bloglist, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/rejectedBlogs", method = RequestMethod.GET)
	public ResponseEntity<List<BlogListModel>> listRejectedblogs() {
		List<Blog> blogs = blogDAO.getblogsByStatus("REJECT");
		List<BlogListModel> bloglist = new ArrayList<BlogListModel>();

		BlogListModel blogModel = null;

		for (Blog b : blogs) {
			blogModel = new BlogListModel();
			blogModel.setBlog(b);
			blogModel.setFirstName(userDAO.getById(b.getUserId()).getFirstName());
			blogModel.setLastname(userDAO.getById(b.getUserId()).getLastName());
			bloglist.add(blogModel);

		}

		if (bloglist.isEmpty()) {
			blog = new Blog();
			blog.setErrorCode("404");
			blog.setErrorMessage("No blogs present.");
			blogs.add(blog);
		}
		return new ResponseEntity<List<BlogListModel>>(bloglist, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/approveBlog/{blogId}", method = RequestMethod.PUT)
	public ResponseEntity<Blog> approveBlog(@PathVariable("blogId") int blogId) {
		blog = blogDAO.get(blogId);
		if (blog == null) {
			blog = new Blog();
			blog.setErrorCode("404");
			blog.setErrorMessage("Invalid blog");
		} else {

			blog.setStatus("APPROVE");
			if (blogDAO.saveOrUpdate(blog) == false) {
				blog = new Blog();
				blog.setErrorCode("404");
				blog.setErrorMessage("Failed to update blog.");
			} else {
				blog.setErrorCode("200");
				blog.setErrorMessage("blog updated successfully.");
			}

		}
		return new ResponseEntity<Blog>(blog, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/disapproveBlog/{blogId}", method = RequestMethod.PUT)
	public ResponseEntity<Blog> disapproveBlog(@PathVariable("blogId") int blogId) {
		blog = blogDAO.get(blogId);
		if (blog == null) {
			blog = new Blog();
			blog.setErrorCode("404");
			blog.setErrorMessage("Invalid blog");
		} else {

			blog.setStatus("REJECT");
			if (blogDAO.saveOrUpdate(blog) == false) {
				blog = new Blog();
				blog.setErrorCode("404");
				blog.setErrorMessage("Failed to update blog.");
			} else {
				blog.setErrorCode("200");
				blog.setErrorMessage("blog updated successfully.");
			}

		}
		return new ResponseEntity<Blog>(blog, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/users", method = RequestMethod.GET)
	public ResponseEntity<List<User>> listAllusers() {
		List<User> users = userDAO.getUsersByStatus("PENDING");
		if (users.isEmpty()) {
			user = new User();
			user.setErrorCode("404");
			user.setErrorMessage("No users present.");
			users.add(user);
		}
		return new ResponseEntity<List<User>>(users, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/approveUser/{userId}", method = RequestMethod.PUT)
	public ResponseEntity<User> approveUser(@PathVariable("userId") int userId) {
		user = userDAO.getById(userId);
		if (user == null) {
			user = new User();
			user.setErrorCode("404");
			user.setErrorMessage("Invalid user");
		} else {

			user.setStatus("APPROVE");
			if (userDAO.saveOrUpdate(user) == false) {
				user = new User();
				user.setErrorCode("404");
				user.setErrorMessage("Failed to update user.");
			} else {
				user.setErrorCode("200");
				user.setErrorMessage("user updated successfully.");
				emailService.approvedUserMessage(user);
			}

		}
		return new ResponseEntity<User>(user, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/disapproveUser/{userId}", method = RequestMethod.PUT)
	public ResponseEntity<User> disapproveUser(@PathVariable("userId") int userId) {
		user = userDAO.getById(userId);
		if (user == null) {
			user = new User();
			user.setErrorCode("404");
			user.setErrorMessage("Invalid user");
		} else {

			user.setStatus("REJECT");
			if (userDAO.saveOrUpdate(user) == false) {
				user = new User();
				user.setErrorCode("404");
				user.setErrorMessage("Failed to update user.");
			} else {
				user.setErrorCode("200");
				user.setErrorMessage("user updated successfully.");
			}

		}
		return new ResponseEntity<User>(user, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/approvedusers", method = RequestMethod.GET)
	public ResponseEntity<List<User>> listApprovedusers() {
		List<User> users = userDAO.getUsersByStatus("APPROVE");

		if (users.isEmpty()) {
			user = new User();
			user.setErrorCode("404");
			user.setErrorMessage("No users present.");
			users.add(user);
		}
		return new ResponseEntity<List<User>>(users, HttpStatus.OK);
	}

	@RequestMapping(value = "/latestusers", method = RequestMethod.GET)
	public ResponseEntity<List<User>> listLatestusers() {
		List<User> users = userDAO.getTopUsers(3);

		if (users.isEmpty()) {
			user = new User();
			user.setErrorCode("404");
			user.setErrorMessage("No users present.");
			users.add(user);
		}
		return new ResponseEntity<List<User>>(users, HttpStatus.OK);
	}

	@RequestMapping(value = "/admin/blockUser/{userId}", method = RequestMethod.PUT)
	public ResponseEntity<User> blockUser(@PathVariable("userId") int userId) {
		user = userDAO.getById(userId);
		if (user == null) {
			user = new User();
			user.setErrorCode("404");
			user.setErrorMessage("Invalid user");
		} else {

			user.setStatus("REJECT");
			user.setEnabled("FALSE");
			if (userDAO.saveOrUpdate(user) == false) {
				user = new User();
				user.setErrorCode("404");
				user.setErrorMessage("Failed to update user.");
			} else {
				user.setErrorCode("200");
				user.setErrorMessage("user updated successfully.");
				emailService.approvedUserMessage(user);
			}

		}
		return new ResponseEntity<User>(user, HttpStatus.OK);
	}

}

File: online_collaboration/app/components/admin/adminController.js
AdminModule.controller('AdminController', ['adminFactory', 'UserFactory', '$rootScope', function (adminFactory, UserFactory, $rootScope) {

    var self = this;
    self.friendId = {};
    self.blogs = [];
    self.blog = {};
    self.users = [];
    self.clients = [];
    self.request = {};
    self.user = {};
    self.none = false;

    function getUnapprovedBlogs() {
        adminFactory.getUnapprovedBlogs().
            then(function (data) {
                console.log(data);
                self.blogs = data;
                if (self.blogs.length <= 0) {
                    self.none = true;
                }
                self.failed = false;
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
            });

    }

    getUnapprovedBlogs();


    self.approveBlog = function (blogId) {
        console.log(blogId);
        adminFactory.approveBlog(blogId)
            .then(function (data) {
                self.blog = data;
                self.failed = false;
                console.log('approval successful');
                getUnapprovedBlogs();
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
                getUnapprovedBlogs();
            });

    }

    self.disapproveBlog = function (blogId) {
        console.log(blogId);
        adminFactory.disapproveBlog(blogId)
            .then(function (data) {
                self.blog = data;
                self.failed = false;
                console.log('disapproval successful');
                getUnapprovedBlogs();
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
                getUnapprovedBlogs();
            });
    }

    function getUnapprovedUsers() {
        adminFactory.getUnapprovedUsers().
            then(function (data) {
                console.log(data);
                self.users = data;
                if (self.users.length <= 0) {
                    self.none = true;
                }
                self.failed = false;
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
            });

    }

    getUnapprovedUsers();

    function getApprovedUsers() {
        adminFactory.getApprovedUsers().
            then(function (data) {
                console.log(data);
                self.clients = data;
                if (self.clients.length <= 0) {
                    self.none = true;
                }
                self.failed = false;
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
            });

    }

    getApprovedUsers();


    self.approveUser = function (userId) {
        console.log(userId);
        adminFactory.approveUser(userId)

            .then(function (data) {
                self.user = data;
                self.failed = false;
                console.log('approval successful');
                getUnapprovedUsers();
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
                getUnapprovedUsers();
            });

    }

    self.disapproveUser = function (userId) {
        console.log(userId);
        adminFactory.disapproveUser(userId)
            .then(function (data) {
                self.user = data;
                self.failed = false;
                console.log('disapproval successful');
                getUnapprovedUsers();
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
                getUnapprovedUsers();
            });
    }

    self.blockUser = function (userId) {
        console.log(userId);
        adminFactory.blockUser(userId)

            .then(function (data) {
                self.user = data;
                self.failed = false;
                console.log('approval successful');
                getApprovedUsers();
            }, function (errResponse) {
                console.error(errResponse);
                self.failed = true;
                getApprovedUsers();
            });

    }

    self.addFriend = function(userId) {
        self.request.friendId = userId;
        console.log($rootScope.user.userId);
        console.log(self.request.friendId);
        self.request.userId = $rootScope.userId;
        console.log(self.request.userId);
        UserFactory.addFriend(self.request)
            .then(
            function (d) {
                self.request = d;
                console.log(self.user);
                getApprovedUsers();
            },
            function (errResponse) {
                console.error('Error sending request User');
            }
            );
    }

}]);
